src/plugins/discord/mirror.js

```diff
@@ -10,18 +10,23 @@ import * as nullUtil from "../../util/null";
  * Mirrors data from the Discord API into a local sqlite db.
  */
 
+const REFETCH = 50;
+
 export async function fetchDiscord(
   sqliteMirror: SqliteMirror,
-  streams: DepaginatedFetcher
+  streams: DepaginatedFetcher,
+  refetchNMessages: number = REFETCH
```

Comment:
It's not immediately obvious to me why we'd want to refetch messages, so it probably won't be obvious to future readers of this code either--could you please add a docstring or comment explaining why we have this parameter? is it based on the premise that recent messages may have had new reactions added?

if so, maybe instead of refetching a fixed # of messages, we should do it via timestamp? i.e. going back 50 messages might only go back a few minutes of history in a busy channel, maybe we should instead refetch all messages that are less than a day old? i recognize this could add more complexity.

Reply:
also, a nit: prefer `nMessagesToRefetch` vs `refetchNMessages`, usually when using `n` to indicate a numeric quantity, we put it at the front. having `refetchNMessages` makes me think that we are refetching a "NMessage", whatever that is 😝

Reply:
My reasoning for fetching based on a fixed message number is that people will be more likely to update or react to posts that are visible, i.e. near the end of their feed, and that a fixed number would capture recent messages, but also older messages on less busy channels.

I'm envisioning a timestamp based approach as trying to capture the case where a user has 50+ new messages in a channel and they scroll through them and add reactions to some of them. I don't personally go through every unread message on a channel, so I'm projecting that on our users and suggesting it's a less probable case. But lmk if that's a flawed assumption. l don't think using a timestamp would add significant complexity, was actually doing that in an iteration with Beanow already.

Reply:
That makes sense, thanks for the explanation @BrianLitwin! Will merge.